server/services/licenceService.ts

```diff
@@ -160,27 +160,29 @@ export default class LicenceService {
     const token = await this.hmppsAuthClient.getSystemClientToken(username)
 
     const requestBody = {
-      data: Object.keys(formData).flatMap((key, index) => {
-        // The POST request to the API will only accept an array of objects where value is a string.
-        // Therefore, if the type of data entered from the form is an array or an object, we need to convert that to a string type.
-        // For arrays, we can just split it into multiple objects each with a value from each member of that array.
-        // For objects, the objects needs to be converted into a string representation. Types which need to be represented as a string
-        // should extend Stringable and implement the stringify() method
+      data: Object.keys(formData)
+        .filter(key => formData[key])
```

Comment:
Filtering out any undefined fields (i.e. where fields are hidden under a conditional radio button which has not been selected)